### PR TITLE
Fix DATE / TIME argument parsing for planetscale (Rust side)

### PR DESCRIPTION
**Pull request by miguelff**
Server side changes for https://github.com/prisma/team-orm/issues/517

```
 PASS  tests/functional/issues/6578/tests.ts
  issues.6578 (provider=sqlite)
    ○ skipped should assert Dates, DateTimes, Times and UUIDs are wrapped in quotes and are deserializable
  issues.6578 (provider=sqlite, js_libsql)
    ○ skipped should assert Dates, DateTimes, Times and UUIDs are wrapped in quotes and are deserializable
  issues.6578 (provider=postgresql)
    ○ skipped should assert Dates, DateTimes, Times and UUIDs are wrapped in quotes and are deserializable
  issues.6578 (provider=postgresql, js_pg)
    ○ skipped should assert Dates, DateTimes, Times and UUIDs are wrapped in quotes and are deserializable
  issues.6578 (provider=postgresql, js_neon)
    ○ skipped should assert Dates, DateTimes, Times and UUIDs are wrapped in quotes and are deserializable
  issues.6578 (provider=mysql)
    ○ skipped should assert Dates, DateTimes, Times and UUIDs are wrapped in quotes and are deserializable
  issues.6578 (provider=mysql, js_planetscale)
    ✓ should assert Dates, DateTimes, Times and UUIDs are wrapped in quotes and are deserializable (144 ms)
  issues.6578 (provider=cockroachdb)
    ○ skipped should assert Dates, DateTimes, Times and UUIDs are wrapped in quotes and are deserializable
  issues.6578 (provider=sqlserver)
    ○ skipped should assert Dates, DateTimes, Times and UUIDs are wrapped in quotes and are deserializable

Test Suites: 1 passed, 1 total
Tests:       8 skipped, 1 passed, 9 total
Snapshots:   0 total
Time:        4.982 s, estimated 6 s
Ran all test suites matching /6578/i.
```

### CODE REVIEW

File: .github/workflows/query-engine-driver-adapters.yml [hunk 1]
```diff
@@ -46,7 +46,7 @@ jobs:
       QUERY_BATCH_SIZE: '10'
       WORKSPACE_ROOT: ${{ github.workspace }}
 
-    runs-on: ubuntu-latest
+    runs-on: buildjet-8vcpu-ubuntu-2004
```
Comment (jkomyno) [inline .github/workflows/query-engine-driver-adapters.yml:49]: Is this change intended?

Comment (janpio) [inline .github/workflows/query-engine-driver-adapters.yml:49]: probably want to undo that before merge

File: query-engine/driver-adapters/src/conversion/mysql.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            None => JsonValue::Null.into(),
+        },
+        quaint::ValueType::DateTime(dt) => match dt {
+            Some(dt) => JSArg::RawString(dt.format(DATE_FORMAT).to_string()),
+            None => JsonValue::Null.into(),
+        },
+        quaint::ValueType::Time(dt) => match dt {
+            Some(dt) => JSArg::RawString(dt.format(TIME_FORMAT).to_string()),
+            None => JsonValue::Null.into(),
+        },
```
Comment (aqrln) [inline query-engine/driver-adapters/src/conversion/mysql.rs:32]: Maybe out of scope of this PR since a lot of this code is moved around as is, and could be changed in a follow-up PR, but I'd get rid of the duplicate `None => JsonValue::Null.into()` match arms and inner `match` statements altogether by making `Some` a part of the outer pattern. The last catch-all pattern will handle `None`s for any `ValueType`.

```suggestion
        quaint::ValueType::Json(Some(s)) => {
            let json_str = serde_json::to_string(s)?;
            JSArg::RawString(json_str)
        }
        quaint::ValueType::Bytes(Some(bytes)) => JSArg::Buffer(bytes.to_vec()),
        quaint::ValueType::Numeric(Some(bd)) => {
            // converting decimal to string to preserve the precision
            JSArg::RawString(bd.to_string())
        }
        quaint::ValueType::DateTime(Some(dt)) => JSArg::RawString(dt.format(DATE_FORMAT).to_string()),
        quaint::ValueType::Time(Some(dt)) => JSArg::RawString(dt.format(TIME_FORMAT).to_string()),
```

File: .github/workflows/query-engine-driver-adapters.yml (hunk 1 above)
Comment (miguelff) [reply .github/workflows/query-engine-driver-adapters.yml:49]: Yes, temporarily. It's subject of each own commit.

File: query-engine/driver-adapters/src/conversion/mysql.rs (hunk 2 above)
Comment (miguelff) [reply query-engine/driver-adapters/src/conversion/mysql.rs:32]: That could be definitely some future improvement.

File: .github/workflows/query-engine-driver-adapters.yml (hunk 1 above)
Comment (miguelff) [reply .github/workflows/query-engine-driver-adapters.yml:49]: Once this is ready to be merged, I will remove that.

**Review by janpio — APPROVED** [review]
Unblocking Miguel here, tests pass and Alberto and Alexey did not have any required changes.

**Review by jkomyno — APPROVED** [review]
Approved with a couple of comments:
- let's remind ourselves to revert the temporary change to `.github/workflows/query-engine-driver-adapters.yml`
- let's either follow up on @aqrln's comment in this PR, or create an issue so we don't forget about it

File: .github/workflows/query-engine-driver-adapters.yml (hunk 1 above)
Comment (janpio) [reply .github/workflows/query-engine-driver-adapters.yml:49]:
```suggestion
    runs-on: ubuntu-latest
```
